src/main/java/org/jabref/gui/fieldeditors/contextmenu/EditorMenus.java

```diff
@@ -84,4 +85,24 @@
             return menuItems;
         };
     }
+
+    /**
+     * The default context menu with a specific menu item to cleanup  URL.
+     *
+     * @param textArea text-area that this menu will be connected to
+     * @return menu containing items of the default menu and an item to cleanup a URL
+     */
+    public static Supplier<List<MenuItem>> getCleanupURLMenu(TextArea textArea) {
+        return () -> {
+            CustomMenuItem cleanupURL = new CustomMenuItem(new Label(Localization.lang("Cleanup URL Link")));
+            //cleanupURL.setDisable(true);
+            cleanupURL.setOnAction(event -> textArea.setText(new CleanupURLFormatter().format(textArea.getText())));
+
+            List<MenuItem> menuItems = new ArrayList<>();
+            menuItems.add(cleanupURL);
+            menuItems.add(new SeparatorMenuItem());
+            menuItems.addAll(getDefaultMenu(textArea).get());
```

Comment:
It should be the second option in your screenshots since "Protect terms" and the other options do not make sense for the URL field.